Eval with retries (#332)
tarantool/ansible-cartridge: doc/eval.md

* [`eval_on_control_instance`](/doc/scenario.md#eval_on_control_instance) runs
  specified snipped only on the control instance.

By default, result is simply saved to `eval_res` variable, but if

> `cartridge_eval_with_retries` is set to `true`, than [eval with retries](#eval-with-retries) is performed.

Reviewer: ```suggestion
`cartridge_eval_with_retries` is set to `true`, than result will be checked for errors, more info [here](#eval-with-retries) is performed.
```

      - age: 24
        city: 'Moscow'
```

## Eval with retries

If `cartridge_eval_with_retries` is set, that `eval` expects that function

> return `res, err`.
> It retries until `err` is received.

Reviewer: ```suggestion
return `res, err`. It retries until `err` is not `nil`.
```

Author: I start each sentence on a new line.

It can be useful to perform some checks in Lua code and wait until some condition is met.

Retries and delay can be configured:

> - `cartridge_eval_retries` (`number`, default: `3`) number of eval retries;
> - `cartridge_eval_delay` (`number`, default: `5`)  - eval retries delay;

Reviewer: ```suggestion
- `cartridge_eval_retries` (`number`, default: `3`) - number of eval retries;
- `cartridge_eval_delay` (`number`, default: `5`) - eval retries delay;
```

```yaml
- name: Eval with retries
  hosts: cluster
  roles:
    - ansible-cartridge
  become: true
  become_user: root
  gather_facts: false
  vars:
    cartridge_scenario:
      - eval
    cartridge_eval_with_retries: true
    cartridge_eval_retries: 10
    cartridge_eval_delay: 1
    cartridge_eval_body: |
      local checkpoint_is_in_progress = box.info().gc().checkpoint_is_in_progress
      if checkpoint_is_in_progress then
        return nil, "Checkpoint is still in progress"
      end

>       return true

Reviewer: Maybe add some info in result? To show that eval_res is defined when retries.

```suggestion
      return ['connector', 'core']
```

Author: `return ['connector', 'core']` would fail. But I will think about it, `res, err` can be shown here.

Author: I can't think of any cases with using `eval_res` =(
